components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.action
+
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.ContentState
+import mozilla.components.browser.state.state.SecurityInfoState
+import mozilla.components.browser.state.state.SessionState
+import mozilla.components.browser.state.state.CustomTabSessionState
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.lib.state.Action
+
+/**
+ * [Action] implementation related to [BrowserState].
+ */
+sealed class BrowserAction : Action
+
+/**
+ * [BrowserAction] implementations related to updating the list of [TabSessionState] inside [BrowserState].
+ */
+sealed class TabsListAction : BrowserAction() {
```

Comment:
nit: `TabListAction` instead of plural `TabsListAction`, same as `CustomTabListAction`?